### PR TITLE
Add Nunjucks highlighting

### PR DESCRIPTION
**Pull request by SimenB**
Search results: https://github.com/search?utf8=%E2%9C%93&q=extension%3Anjk+NOT+nothack&type=Code&ref=searchresults

Nunjucks docs on file extension: https://mozilla.github.io/nunjucks/templating.html#file-extensions

> # File extensions
> Although you are free to use any file extension you wish for your Nunjucks template files, the Nunjucks community has adopted `.njk`.
> If you are developing tools or editor syntax helpers for Nunjucks, please include recognition of the `.njk` extension.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -1659,13 +1659,16 @@ HTML+Django:
   tm_scope: text.html.django
   group: HTML
   extensions:
-  - ".mustache"
   - ".jinja"
+  - ".mustache"
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Why did you move the `.mustache` extension?

File: samples/HTML+Django/nunjucks.njk [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{% extends "parent.html" %}
```
Comment (pchaigno) [inline samples/HTML+Django/nunjucks.njk]: Is this from a "real" application? Do you have a more realistic sample file? Sample files are used to train our Bayesian classifier so it's important they reflect in-the-wild syntaxes and keywords.

File: lib/linguist/languages.yml (hunk 1 above)
Comment (SimenB) [inline lib/linguist/languages.yml]: To make it alphabetical. The tests told me off since i put `njk` before `jinja`, so i just sorted the whole thing 

https://travis-ci.org/github/linguist/jobs/178098254 (line 1108)

File: samples/HTML+Django/nunjucks.njk (hunk 2 above)
Comment (SimenB) [inline samples/HTML+Django/nunjucks.njk]: No, just a random snippet. I can add a more substantial example, though!

File: lib/linguist/languages.yml (hunk 1 above)
Comment (Alhadis) [inline lib/linguist/languages.yml]: Ah, please don't. We make an exception for the first extension in an extensions list, because that's what marks the "primary" (most common) extension. So what you've done here is make `.jinja` the primary extension of `HTML+Django`.

Comment (SimenB) [inline lib/linguist/languages.yml]: Well, that's true, isn't it? Mustache is primarily a js templating language. And the django just says "django templating language" (no mention of Mustache at all), and calls out jinja2 as an alternative. 

https://docs.djangoproject.com/en/1.10/topics/templates/

File: samples/HTML+Django/nunjucks.njk (hunk 2 above)
Comment (pchaigno) [inline samples/HTML+Django/nunjucks.njk]: That'd be great!

**Comment by SimenB** [comment]
@pchaigno Added example with `extends`, `macro`, `if`, `block, `for`, comments, a filter, and `import`. Mostly cut from https://mozilla.github.io/nunjucks/templating.html

**Comment by SimenB** [comment]
@pchaigno Is this GTG?

**Comment by brandonblack** [comment]
Thanks @SimenB! 🎉
